tests/src/main/java/net/neoforged/neoforge/debug/client/DimensionTransitionTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.debug.client;
+
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.neoforged.api.distmarker.Dist;
+import net.neoforged.neoforge.client.DimensionTransitionScreen;
+import net.neoforged.neoforge.client.event.RegisterDimensionTransitionEvent;
+import net.neoforged.neoforge.event.entity.player.PlayerEvent;
+import net.neoforged.testframework.DynamicTest;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+
+@ForEachTest(side = Dist.CLIENT, groups = DimensionTransitionTests.GROUP)
+public class DimensionTransitionTests {
+    public static final String GROUP = "dimension_transition";
+    public static final ResourceLocation NETHER_BG = ResourceLocation.withDefaultNamespace("textures/block/netherrack.png");
+    public static final ResourceLocation END_BG = ResourceLocation.withDefaultNamespace("textures/block/end_stone.png");
+
+    @EmptyTemplate
+    @TestHolder(description = "Tests if a custom dimension transition is properly displayed", enabledByDefault = true)
+    static void dimensionTransitions(DynamicTest test) {
```

Comment:
For completeness, this test should be split into two: one for the end dimension outgoing transition screen, and one for the nether dimension incoming transition screen. 

Currently, because the two are combined, confirming the success of one screen confirms the whole test, even if the other screen hasn't been tested yet.